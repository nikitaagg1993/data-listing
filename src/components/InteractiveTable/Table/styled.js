import styled, { css } from 'styled-components';

const disabled = css`
  pointer-events: none;
  opacity: 0.4;
`;

const TableStyle = styled.div`
  position: relative;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  ${(props) => props.disabled && disabled};
  ${(props) => props.theme.table.style};
`;

export const ResponsiveWrapper = styled.div`
  position: relative;
  width: 100%;
  border-radius: inherit;
  ${(props) => props.responsive && css`
    overflow-x: auto;

    // prevents vertical scrolling in firefox
    overflow-y: hidden;
    min-height: 0;
  `};
  ${(props) => props.overflowY && props.responsive && props.overflowYOffset && css`
    padding-bottom: ${props.overflowYOffset};
    margin-bottom: -${props.overflowYOffset};
  `};
`;

export const TableBody = styled.div`
  display: flex;
  flex-direction: column;
  ${({
    fixedHeader, hasOffset, offset, fixedHeaderScrollHeight,
  }) => fixedHeader && css`
    max-height: ${hasOffset ? `calc(${fixedHeaderScrollHeight} - ${offset})` : fixedHeaderScrollHeight};
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
  `};
`;

export const TableHead = styled.div`
  display: flex;
  ${(props) => props.theme.head.style};
`;

const disabledTableHeadRow = css`
  pointer-events: none;
`;

export const TableHeadRow = styled.div`
  display: flex;
  align-items: stretch;
  width: 100%;
  ${(props) => props.theme.headRow.style};
  ${(props) => props.disabled && disabledTableHeadRow};
`;

export const TableWrapper = styled.div`
  position: relative;
  width: 95%;
  border-style: groove;
  border-width: 5px;
  margin: 10px;
  padding: 10px;
  align-self: center;
  border-color: lightgray;
  ${(props) => props.theme.tableWrapper.style};
`;

export default TableStyle;
