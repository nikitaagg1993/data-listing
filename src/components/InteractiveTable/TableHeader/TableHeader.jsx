import React from 'react';
import PropTypes from 'prop-types';
import TableHeaderStyle, { Title } from './styled';
// import ContextMenu from './ContextMenu';

const TableHeader = ({ title }) => (
  <TableHeaderStyle>
    <Title>
      {title}
    </Title>
  </TableHeaderStyle>
);

TableHeader.propTypes = {
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
};

export default TableHeader;
