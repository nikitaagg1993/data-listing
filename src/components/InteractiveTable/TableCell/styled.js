import styled, { css } from 'styled-components';
// import { Cell } from './Cell';
import { media } from '../../shared/media';

const overflowCSS = css`
  div:first-child {
    white-space: ${(props) => (props.column.wrap ? 'normal' : 'nowrap')};
    overflow: ${(props) => (props.column.allowOverflow ? 'visible' : 'hidden')};
    text-overflow: ellipsis;
  }
`;

export const CellBase = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  ${({ theme, head }) => theme && theme[head ? 'headCells' : 'cells'] && theme[head ? 'headCells' : 'cells'].style};
  ${(props) => props.noPadding && 'padding: 0'};
`;

// Flex calculations
export const Cell = styled(CellBase)`
 

  ${(props) => props.column.width && css`
     width: ${props.column.width};
  `};

  ${(props) => !props.column.width && css`
    flex-grow: 1;
    flex-shrink: 0;
    flex-basis: 0;
  `};

  ${(props) => props.column.numeric && css`
    justify-content: flex-end;
  `};
  ${(props) => props.column.right && 'justify-content: flex-end'};
  ${(props) => (props.column.center || props.column.button) && 'justify-content: center'};
  ${(props) => (props.column.compact || props.column.button) && 'padding: 0'};

  /* handle hiding cells */
  ${(props) => props.column.hide && props.column.hide === 'sm' && media.sm`
    display: none;
  `};
  ${(props) => props.column.hide && props.column.hide === 'md' && media.md`
    display: none;
  `};
  ${(props) => props.column.hide && props.column.hide === 'lg' && media.lg`
    display: none;
  `};
  ${(props) => props.column.hide && Number.isInteger(props.column.hide) && media.custom(props.column.hide)`
    display: none;
  `};
`;

const TableCellStyle = styled(Cell)`
  font-size: ${(props) => props.theme.rows && props.theme.rows.fontSize};
  font-weight: 400;
  ${(props) => !props.column.cell && overflowCSS};
  ${(props) => props.column.style};
  ${(props) => props.extendedCellStyle};
`;

export default TableCellStyle;
