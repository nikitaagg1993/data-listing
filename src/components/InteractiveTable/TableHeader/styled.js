import styled from 'styled-components';

const TableHeaderStyle = styled.header`
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  display: flex;
  flex: 1 1 auto;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  flex-wrap: wrap;
  ${(props) => props.theme.heade && props.theme.header.style}
`;

export const Title = styled.div`
  flex: 1 0 auto;
  color: ${(props) => props.theme.header && props.theme.header.fontColor};
  font-size: ${(props) => props.theme.header && props.theme.header.fontSize};
  font-weight: 400;
  text-align: center;
`;

export default TableHeaderStyle;
