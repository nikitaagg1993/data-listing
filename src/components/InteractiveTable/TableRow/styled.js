import styled from 'styled-components';

const TableRowStyle = styled.div`
  display: flex;
  align-items: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  ${(props) => props.theme.rows && props.theme.rows.style};
  ${(props) => props.theme.rows && props.selected && props.theme.rows.selectedHighlightStyle};
  ${(props) => props.extendedRowStyle};
`;

export default TableRowStyle;
