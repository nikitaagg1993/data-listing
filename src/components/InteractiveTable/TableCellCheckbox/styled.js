import styled from 'styled-components';
import { CellBase } from '../TableCell/styled';

const TableCellCheckboxStyle = styled(CellBase)`
  flex: 0 0 48px;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
`;

export default TableCellCheckboxStyle;
