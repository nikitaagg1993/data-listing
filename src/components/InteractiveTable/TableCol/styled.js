import styled from 'styled-components';
import { Cell } from '../TableCell/styled';

const TableColStyle = styled(Cell)`
  ${(props) => props.column.button && 'text-align: center'};
`;

export const ColumnSortable = styled.div`
  display: inline-flex;
  align-items: center;
  height: 100%;
  line-height: 1;
  user-select: none;
  ${(props) => (props.theme.headCells && (props.sortActive ? props.theme.headCells.activeSortStyle : props.theme.headCells.inactiveSortStyle))};

  &:hover {
    ${({ column }) => column.sortable && 'cursor: pointer'};
    ${({ column, theme }) => column.sortable && theme.headCells && theme.headCells.activeStyle};

    span {
      ${({ sortActive, column }) => !sortActive && column.sortable && 'opacity: 1'};
    }
  }
`;

export default TableColStyle;
