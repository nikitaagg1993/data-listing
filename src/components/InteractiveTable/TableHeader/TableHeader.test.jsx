import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TableHeader from './TableHeader';
import { Title } from './styled';

jest.mock('../DataTableContext', () => ({
  useTableContext: () => ({
    dispatch: jest.fn(),
    data: [],
  }),
}));

test('Table Col Checkbox', () => {
  const tree = renderer.create(<TableHeader title="Title" />);
  const testInstance = tree.root;

  expect(tree.toJSON()).toMatchSnapshot();
  expect(testInstance.findAllByType(Title).length).toEqual(1);
  expect(testInstance.props).toMatchObject({ title: 'Title' });
});
