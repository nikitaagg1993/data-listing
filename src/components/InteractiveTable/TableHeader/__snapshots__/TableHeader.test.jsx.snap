// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Col Checkbox 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
  overflow: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font-weight: 400;
  text-align: center;
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    Title
  </div>
</header>
`;
